File: ido_sdk_flutter/ido_sdk/bluetooth_sdk/android/src/main/kotlin/com/example/flutter_bluetooth/ble/reconnect/BaseLeScanner.java
package com.example.flutter_bluetooth.ble.reconnect;

import android.bluetooth.BluetoothDevice;

public abstract class BaseLeScanner {
    public abstract void startLeScan();

    public abstract void stopLeScan();

    public abstract void callbackOnLeScan(final BluetoothDevice device, final int rssi, final byte[] scanRecord);
}

File: ido_sdk_flutter/ido_sdk/bluetooth_sdk/android/src/main/kotlin/com/example/flutter_bluetooth/ble/callback/BaseLeScanner.kt
package com.example.flutter_bluetooth.ble.callback

import com.example.flutter_bluetooth.ble.scan.ScanLeCallBack

abstract class BaseLeScanner {
    abstract fun startLeScan(data: Map<String, Any?>?, scanLeCallback: ScanLeCallBack?)
    abstract fun stopLeScan()
    abstract fun callbackOnLeScan(scanResult: Map<String, Any?>)
}